microbial_thermodynamics/scipy_demo.py

```diff
@@ -50,10 +49,20 @@ def full_equation_set(
         t,
         a=y[2 * number_of_species : 3 * number_of_species],
         R=y[number_of_species : 2 * number_of_species],
+        s=y[3 * number_of_species],
+        w=y[3 * number_of_species + 1],
+        reaction_energy=reaction_energies,
+        v=[1.0, 2.0],
```

Comment:
If you change this (and the one below) to `v=np.array([1.0, 2.0])` does that work

Reply:
I think it might work with a `numpy` array, but doesn't work with lists of floats

Reply:
No I still get the message TypeError: unsupported operand type(s) for -: 'float' and 'list'

Reply:
Does it work when you change the type hint for v to `NDArray[np.float32]`?

Reply:
I currently have the type hint for v and Reaction_energy set to NDArray[np.float32]. I have pushed what it currently looks like.